tests/periph_i2c/main.c

```diff
@@ -20,270 +20,547 @@
-        puts("          1 -> SPEED_NORMAL (100kbit/s)");
-        puts("          2 -> SPEED_FAST (400kbit/s)");
-        puts("          3 -> SPEED_FAST_PLUS (1Mbit/s)");
-        puts("          4 -> SPEED_HIGH (3.4Mbit/s)\n");
         return 1;
     }
-
     dev = atoi(argv[1]);
-    speed = atoi(argv[2]);
+    if (dev < 0 || dev >= (int)I2C_NUMOF) {
+        puts("Error: Given device not available");
+        i2c_dev = -1;
+        return 1;
+    }
+    else {
+        printf("Success: i2c_%i successfully assigned as master\n", dev);
+        i2c_dev = dev;
+        return 0;
+    }
+    return 1;
+}
+
+int cmd_i2c_acquire(int argc, char **argv) {
+    int res;
+    (void)argv;
+
+    if (check_dev() == 1){
+        return 1;
+    }
+
+    if (argc != 1) {
+        INVALID_ARGS;
+        return 1;
+    }
+
+    printf("Command: i2c_acquire(i2c_%i)\n", i2c_dev);
+    res = i2c_acquire(i2c_dev);
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i successfully acquired\n", i2c_dev);
+        return 0;
+    }
+    return print_i2c_error(res);
+}
 
-    if (speed < 0 || speed > 4) {
-        puts("Error: Speed value not in range (0 - 4)");
+int cmd_i2c_release(int argc, char **argv) {
+    (void)argv;
+    int res;
+
+    if (check_dev() == 1){
         return 1;
     }
 
-    res = i2c_init_master(dev, map_speed[speed]);
-    if (res == -1) {
-        puts("Error: Init: Given device not available");
+    if (argc != 1) {
+        INVALID_ARGS;
         return 1;
     }
-    else if (res == -2) {
-        puts("Error: Init: Unsupported speed value");
+
+    printf("Command: i2c_release(i2c_%i)\n", i2c_dev);
+    res = i2c_release(i2c_dev);
+
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i successfully released\n", i2c_dev);
+        return 0;
+    }
+    return print_i2c_error(res);
+}
+
+int cmd_i2c_read_reg(int argc, char **argv) {
+    int res;
+    uint8_t addr;
+    uint8_t reg;
+    uint8_t flags = 0;
+    uint8_t data;
+
+    if (check_dev() == 1){
         return 1;
     }
-    else {
-        printf("I2C_%i successfully initialized as master!\n", dev);
-        i2c_dev = dev;
+
+    if (argc < 3 || argc > 4) {
+        INVALID_ARGS;
+        printf("Usage:\n%s ADDR REG [FLAG]\n", argv[0]);
+        return 1;
     }
 
-    return 0;
+    addr = atoi(argv[1]);
+    reg = atoi(argv[2]);
+    if (argc == 4) {
+        flags = atoi(argv[3]);
+    }
+
+    printf("Command: i2c_read_reg(i2c_%i, 0x%02x, 0x%02x)\n", i2c_dev, addr, reg);
+    res = i2c_read_reg(i2c_dev, addr, reg, &data, flags);
+
+
+    if (res == I2C_ACK) {
+        printf("Success: i2c_%i successfully read 1 byte from reg 0x%02x: [",
+            i2c_dev, reg);
+        printf("0x%02x", data);
+        puts("]");
+        return 0;
+    }
+    return print_i2c_error(res);
 }
 
-int cmd_write(int argc, char **argv)
-{
+int cmd_i2c_read_regs(int argc, char **argv) {
     int res;
     uint8_t addr;
-    int length = argc - 2;
+    uint8_t reg;
```

Comment:
is this enough, because the API uses `uint16_t`, or not?

Reply:
Yup, legacy stuff [TODO]